aio_config/router_config.md

### 1. v2ray启动

```
#!/bin/sh /etc/rc.common

USE_PROCD=1
START=95
STOP=15
#STOP_CMD="cat /var/run/xray.pid | xargs kill -9"
START_CMD="/root/passwall/xray/xray run -config /root/passwall/xray/config.json"

start_service() {
    echo "start xray"
    procd_open_instance xray # 给服务实例定义一个名称
    procd_set_param command $START_CMD # 需要在前台被执行的服务
    # procd_append_param command -bar 42 # 给以上命令附加的指令参数

    # 如果服务意外中止了，定义 redpawn 可以自动重启它，如果服务命令的确只需要运行一次，需要谨慎设定这里
    # 如果进程在 respawn_threshold 定义的时间内结束了，则判定为进程崩溃并尝试重启它，尝试5次后会停止重启
    procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}

    procd_set_param env SOME_VARIABLE=funtimes  # 给进程传递环境变量
    # procd_set_param limits core="unlimited"  # If you need to set ulimit for your process
    # procd_set_param file /var/etc/your_service.conf # 如果此处定义的文件发生了变化，则会触发 /etc/init.d/your_service reload 重启进程
    # procd_set_param netdev dev # likewise, except if dev's ifindex changes.
    # procd_set_param data name=value ... # likewise, except if this data changes.
    procd_set_param stdout 1 # 转发 stdout 输出到 logd
    procd_set_param stderr 1 # same for stderr
    procd_set_param user root # 以 nobody 用户运行服务
    procd_set_param pidfile /var/run/xray.pid # 在服务启动时写入一个 pid 文件，在停止服务时删除此 pid 文件
    procd_close_instance # 结束服务实例配置
}

stop_service() {
    echo "stop xray"
    cat /var/run/xray.pid | xargs kill -9
}

restart_service() {
    stop
    start
}
```

### 2. 启动命令
```
cp /root/passwall/gfw_custom.conf /tmp/dnsmasq.d/
cp /root/passwall/gfwlist.conf /tmp/dnsmasq.d/

/etc/init.d/dnsmasq restart

ipset create gfwlist hash:net timeout 3600 hashsize 10000 maxelem 100000
# ipset add gfwlist 198.18.0.0/15
ipset add gfwlist 91.108.4.0/22 timeout 0
ipset add gfwlist 91.108.8.0/22 timeout 0
ipset add gfwlist 91.108.12.0/22 timeout 0
ipset add gfwlist 91.108.20.0/22 timeout 0
ipset add gfwlist 91.108.36.0/23 timeout 0
ipset add gfwlist 91.108.38.0/23 timeout 0
ipset add gfwlist 91.108.56.0/22 timeout 0
ipset add gfwlist 149.154.160.0/20 timeout 0
ipset add gfwlist 149.154.164.0/22 timeout 0
ipset add gfwlist 149.154.172.0/22 timeout 0
ipset add gfwlist 185.76.151.0/24 timeout 0

iptables -t nat -N V2RAY
iptables -t nat -A V2RAY -j RETURN -m mark --mark 0xff
iptables -t nat -A V2RAY -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070
iptables -t nat -A PREROUTING -j V2RAY


iptables -t nat -N V2RAY_MASK
iptables -t nat -A V2RAY_MASK -j RETURN -m mark --mark 0xff
iptables -t nat -A V2RAY_MASK -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070
iptables -t nat -A OUTPUT -j V2RAY_MASK

ipset create gfwlist6 hash:net family inet6 timeout 3600 hashsize 10000 maxelem 100000
ipset add gfwlist6 2001:b28:f23d::/48 timeout 0
ipset add gfwlist6 2001:b28:f23f::/48 timeout 0
ipset add gfwlist6 2001:67c:4e8::/48 timeout 0
ipset add gfwlist6 2001:b28:f23c::/48 timeout 0
ipset add gfwlist6 2a0a:f280::/32 timeout 0

ip6tables -t nat -N V2RAY6
ip6tables -t nat -A V2RAY6 -j RETURN -m mark --mark 0xff
ip6tables -t nat -A V2RAY6 -p tcp -m set --match-set gfwlist6 dst -j REDIRECT --to-port 1070
ip6tables -t nat -A PREROUTING -j V2RAY6


ip6tables -t nat -N V2RAY_MASK6
ip6tables -t nat -A V2RAY_MASK6 -j RETURN -m mark --mark 0xff
ip6tables -t nat -A V2RAY_MASK6 -p tcp -m set --match-set gfwlist6 dst -j REDIRECT --to-port 1070
ip6tables -t nat -A OUTPUT -j V2RAY_MASK6

/etc/init.d/ddns restart
/etc/init.d/nlbwmon restart

```

### 3. 小米路由器旁路由配置
```
echo 0 > /proc/sys/net/bridge/bridge-nf-call-iptables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-ip6tables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-arptables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-custom
```

### 4. 重装流程
#### 1. 备份
1. 宽带账号密码
261010087356
33848492
2. v2ray配置
#### 2. 安装
1. 替换镜像, 然后恢复出厂


### 5. warp安装说明
```
https://github.com/fscarmen/warp
https://github.com/pufferffish/wireproxy

warp优选:
https://blog.misaka.rest/2023/03/12/cf-warp-yxip/

workers创建vless协议
https://jdssl.top/index.php/2023/07/21/2023vpn/?__cf_chl_tk=5b2rH74paFlUZYNMmBiBRT515ghCbt7DgdtGGjYzoOg-1693450307-0-gaNycGzNCvs
```

### gfw定时更新
```
```

### 测速地址
```shell
https://gh.con.sh/https://github.com/AaronFeng753/Waifu2x-Extension-GUI/releases/download/v2.21.12/Waifu2x-Extension-GUI-v2.21.12-Portable.7z
https://gh.api.99988866.xyz/https://github.com/AaronFeng753/Waifu2x-Extension-GUI/releases/download/v2.21.12/Waifu2x-Extension-GUI-v2.21.12-Portable.7z
http://gh.msx.workers.dev/https://github.com/AaronFeng753/Waifu2x-Extension-GUI/releases/download/v2.21.12/Waifu2x-Extension-GUI-v2.21.12-Portable.7z
https://github.91chifun.workers.dev/https://github.com/AaronFeng753/Waifu2x-Extension-GUI/releases/download/v2.21.12/Waifu2x-Extension-GUI-v2.21.12-Portable.7z
https://gh-proxy.henryjiu.workers.dev/https://github.com/AaronFeng753/Waifu2x-Extension-GUI/releases/download/v2.21.12/Waifu2x-Extension-GUI-v2.21.12-Portable.7z

```

openwrt_v2ray/router_config.md

### 1. v2ray启动

```
#!/bin/sh /etc/rc.common

USE_PROCD=1
START=95
STOP=15

start_service() {
    procd_open_instance v2ray # 给服务实例定义一个名称
    procd_set_param command /root/passwall/v2ray/v2ray run -config /root/passwall/v2ray/config.json # 需要在前台被执行的服务
    # procd_append_param command -bar 42 # 给以上命令附加的指令参数

    # 如果服务意外中止了，定义 redpawn 可以自动重启它，如果服务命令的确只需要运行一次，需要谨慎设定这里
    # 如果进程在 respawn_threshold 定义的时间内结束了，则判定为进程崩溃并尝试重启它，尝试5次后会停止重启
    procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}

    procd_set_param env SOME_VARIABLE=funtimes  # 给进程传递环境变量
    procd_set_param limits core="unlimited"  # If you need to set ulimit for your process
    # procd_set_param file /var/etc/your_service.conf # 如果此处定义的文件发生了变化，则会触发 /etc/init.d/your_service reload 重启进程
    procd_set_param netdev dev # likewise, except if dev's ifindex changes.
    procd_set_param data name=value ... # likewise, except if this data changes.
    procd_set_param stdout 1 # 转发 stdout 输出到 logd
    procd_set_param stderr 1 # same for stderr
    procd_set_param user root # 以 nobody 用户运行服务
    procd_set_param pidfile /var/run/somefile.pid # 在服务启动时写入一个 pid 文件，在停止服务时删除此 pid 文件
    procd_close_instance # 结束服务实例配置
}
stop_service() {
    ps | grep v2ray | grep -v grep | awk '{print $1}' | xargs kill -9
}
restart_service() {
    stop
    start
}
```

### 2. 启动命令
```
cp /root/passwall/gfw_custom.conf /tmp/dnsmasq.d/
cp /root/passwall/gfwlist.conf /tmp/dnsmasq.d/

/etc/init.d/dnsmasq restart

ipset create gfwlist hash:net
ipset add gfwlist 198.18.0.0/15
ipset add gfwlist 91.108.4.0/22
ipset add gfwlist 91.108.8.0/22
ipset add gfwlist 91.108.12.0/22
ipset add gfwlist 91.108.20.0/22
ipset add gfwlist 91.108.36.0/23
ipset add gfwlist 91.108.38.0/23
ipset add gfwlist 91.108.56.0/22
ipset add gfwlist 149.154.160.0/20
ipset add gfwlist 149.154.164.0/22
ipset add gfwlist 149.154.172.0/22

iptables -t nat -A PREROUTING -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070
iptables -t nat -A OUTPUT -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070
```

### 3. 小米路由器旁路由配置
```
echo 0 > /proc/sys/net/bridge/bridge-nf-call-iptables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-ip6tables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-arptables
echo 0 > /proc/sys/net/bridge/bridge-nf-call-custom
```

openwrt_v2ray/config.sh
# 更新pve源
echo "deb https://mirrors.ustc.edu.cn/proxmox/debian/pve bullseye pve-no-subscription" > /etc/apt/sources.list.d/pve-no-subscription.list
# 删除之前
sed -i.bak "s#ftp.debian.org/debian#mirrors.aliyun.com/debian#g" /etc/apt/sources.list
sed -i "s#security.debian.org#mirrors.aliyun.com/debian-security#g" /etc/apt/sources.list

# 直通
/etc/default/grub
GRUB_CMDLINE_LINUX_DEFAULT="quiet intel_iommu=on"
sed -i 's/quiet/quiet intel_iommu=on/g' /etc/default/grub
echo -e 'vfio\nvfio_iommu_type1\nvfio_pci\nvfio_virqfd' >> /etc/modules

update-grub
update-initramfs -u -k all


cp /root/passwall/gfw_custom.conf /tmp/dnsmasq.d/
cp /root/passwall/gfwlist.conf /tmp/dnsmasq.d/

/etc/init.d/dnsmasq restart

ipset create gfwlist hash:net
ipset add gfwlist 198.18.0.0/15
ipset add gfwlist 91.108.4.0/22
ipset add gfwlist 91.108.8.0/22
ipset add gfwlist 91.108.12.0/22
ipset add gfwlist 91.108.20.0/22
ipset add gfwlist 91.108.36.0/23
ipset add gfwlist 91.108.38.0/23
ipset add gfwlist 91.108.56.0/22
ipset add gfwlist 149.154.160.0/20
ipset add gfwlist 149.154.164.0/22
ipset add gfwlist 149.154.172.0/22

iptables -t nat -A PREROUTING -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070
iptables -t nat -A OUTPUT -p tcp -m set --match-set gfwlist dst -j REDIRECT --to-port 1070




261010087356
33848492


qm set 101 -sata1 /dev/disk/by-id/ata-ST2000DM005-2CW102_ZFM0G0QB

/var/lib/vz/template/iso/openwrt.img

qm importdisk 100 /var/lib/vz/template/iso/openwrt.img local-lvm
qm importdisk 100 /var/lib/vz/template/iso/openwrt-05.09.2023-x86-64-generic-squashfs-combined-efi.img local-lvm
qm importdisk 101 /var/lib/vz/template/iso/arpl.img local-lvm


ip=`curl 6.ipw.cn`; curl "http://dynv6.com/api/update?hostname=openmv.dynv6.net&token=HBsK6jctuYNRDd2pAYaC3-k_CtvxC7&ipv6=$ip"



http://dynv6.com/api/update?hostname=openmv.dynv6.net&token=HBsK6jctuYNRDd2pAYaC3-k_CtvxC7&ipv6=2409:8a00:5469:5bd0:7680:8956:7f7a:74a7

acme.sh --issue --dns dns_dynv6 -d openmv.dynv6.net -d *.openmv.dynv6.net

acme.sh --install-cert -d openmv.dynv6.net \
--key-file       /root/cert/key.pem  \
--fullchain-file /root/cert/cert.pem \
--reloadcmd     "service nginx force-reload"


acme.sh  --remove  -d openmv.dynv6.net

wget -O - https://github.com/OpenMediaVault-Plugin-Developers/packages/raw/master/install | bash


docker run --name webdav --restart always -v /root/webdav:/var/lib/dav -e AUTH_TYPE=Digest -e USERNAME=dujian -e PASSWORD=22372901 --publish 8972:80 -d bytemark/webdav




server {
    # 服务器端口使用443，开启ssl, 这里ssl就是上面安装的ssl模块
    listen       *:443 ssl;
    listen       [::]:1234 ssl;
    # 域名，多个以空格分开
    server_name  openmv.dynv6.net;
    
    # ssl证书地址
    ssl_certificate      /root/cert/cert.pem;  # pem文件的路径
    ssl_certificate_key  /root/cert/key.pem; # key文件的路径
    
    # ssl验证相关配置
    ssl_session_timeout  5m;    #缓存有效期
    ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE:ECDH:AES:HIGH:!NULL:!aNULL:!MD5:!ADH:!RC4;    #加密算法
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;    #安全链接可选的加密协议
    ssl_prefer_server_ciphers on;   #使用服务器端的首选算法

    location /openwrt {
        proxy_http_version 1.1; #代理使用的http协议
        proxy_set_header Host $host; #header添加请求host信息
        proxy_set_header X-Real-IP $remote_addr; # header增加请求来源IP信息
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; # 增加代理记录
        proxy_pass http://192.168.31.1; #服务A访问地址
    }
}
server {
    listen       80; #监听端口
    server_name  openmv.dynv6.net; #请求域名
    return      301 https://$host:1234$request_uri; #重定向至https访问。
}

docker run -d --name=webdav --restart=always -v /path/config.yml:/config.yml -p 8080:8080 stilleshan/webdav


docker run --name some-sftpgo \
    -p 8080:8080 \
    -p 2022:2022 \
    -p 8972:8972 \
    --mount type=bind,source=/root/webdav,target=/srv/sftpgo \
    -e SFTPGO_WEBDAVD__BINDINGS__0__PORT=8972 \
    -d "drakkan/sftpgo"

https://omv.openmv.dynv6.net:1234/


docker run --name webdav \
  --restart=unless-stopped \
  -p 8972:80 \
  -v /root/webdav/dav:/media \
  -e USERNAME=dujian \
  -e PASSWORD=22372901 \
  -e TZ=Asia/Shanghai  \
  -e UDI=1000 \
  -e GID=1000 \
  -d  ugeek/webdav:amd64



0:0:0:0:0:ffff:c0a8:1f05

0000:0000:0000:0000:0000:0000:c0a8:1f05
0:0:0:0:0:0:c0a8:1f05
